src/Compiler/CompilationResult.php

```diff
@@ -0,0 +1,305 @@
+<?php
+
+/**
+ * SCSSPHP
+ *
+ * @copyright 2012-2020 Leaf Corcoran
+ *
+ * @license http://opensource.org/licenses/MIT MIT
+ *
+ * @link http://scssphp.github.io/scssphp
+ */
+
+namespace ScssPhp\ScssPhp\Compiler;
+
+use ScssPhp\ScssPhp\Util;
+use ScssPhp\ScssPhp\Exception\CompilerException;
+
+/**
+ * Compiler environment
+ *
+ */
+class CompilationResult
+{
+    protected static $resolvedImport = [];
+
+    protected $isCached = false;
+
+    /**
+     * @var string
+     */
+    protected $css = '';
+
+    /**
+     * @var string
+     */
+    protected $sourceMap = '';
+
+    protected $sourceMapFile;
+    protected $sourceMapUrl;
+
+
+    /**
+     * All the effective parsedfiles
+     * @var array
+     */
+    protected $parsedFiles = [];
+
+    /**
+     * All the @import files and urls seen in the compilation process
+     * @var array
+     */
+    protected $includedFiles = [];
+
+    /**
+     * All the @import files resolved and imported (use to check the once condition)
+     * @var array
+     */
+    protected $importedFiles = [];
+
+
+    /**
+     * @param bool $isCached
+     */
+    public function setIsCached($isCached) {
+        $this->isCached = $isCached;
+    }
+
+    /**
+     * @return bool
+     */
+    public function getIsCached() {
+        return $this->isCached;
+    }
+
+    /**
+     * @param string $css
+     */
+    public function setCss($css)
+    {
+        $this->css = $css;
+    }
+
+    /**
+     * @param string $css
+     */
+    public function appendCss($css)
+    {
+        $this->css .= $css;
+    }
+
+    /**
+     * @return string
+     */
+    public function getCss()
+    {
+        return $this->css . $this->getSourceMapCss();
+    }
+
+    /**
+     * Adds to list of parsed files
+     *
+     * @api
+     *
+     * @param string $path
+     *
+     * @return void
+     */
+    public function addParsedFile($path)
+    {
+        if (isset($path) && is_file($path)) {
```

Comment:
this `isset` is useless useless if the param is really a `string` and not `string|null`

Reply:
In fact it's `string|null` and I prefer to keep this as is as I don't want to introduce a test here 
https://github.com/scssphp/scssphp/blob/master/src/Compiler.php#L481
which would tend to make someone move the previous line in the if whereas we need to fill the array whatever the value of `$path`

Reply:
please change it to a null check then to be clearer